src/context/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, ReactNode, useContext, FC, useState } from 'react'
+import { useRouter } from 'next/router';
+
+interface Props {
+    children?: ReactNode;
+}
+
+export type ContextType = {
+    state: { toggleEditButton: boolean };
+    actions: { handleToggleButton: () => void }
+}
+const AppContext = createContext<ContextType | null>(null)
+
+const AppWrapper: FC<Props> = ({ children }) => {
```

Comment:
App Wrapper would again be a generic term here, we might face difficulties in future to refer

Reply:
This contextWrapper serves a purpose right.. shouldn't the naming convention be similar, @ankushdharkar wanted your opinion on this